{
  "CVE Code": "CVE-2023-36807",
  "Published": "2023-06-30T18:38:10.303Z",
  "Updated": "2024-11-06T19:51:43.120Z",
  "Description": "pypdf is a pure-python PDF library capable of splitting, merging, cropping, and transforming the pages of PDF files. In version 2.10.5 an attacker who uses this vulnerability can craft a PDF which leads to an infinite loop. This infinite loop blocks the current process and can utilize a single core of the CPU by 100%. It does not affect memory usage. That is, for example, the case if the user extracted metadata from such a malformed PDF. Versions prior to 2.10.5 throw an error, but do not hang forever. This issue was fixed with https://github.com/py-pdf/pypdf/pull/1331 which has been included in release 2.10.6. Users are advised to upgrade. Users unable to upgrade should modify `PyPDF2/generic/_data_structures.py::read_object` to an an error throwing case. See GHSA-hm9v-vj3r-r55m for details. ",
  "ProblemTypes": [
    "CWE-835: Loop with Unreachable Exit Condition ('Infinite Loop')"
  ],
  "Affected": [
    {
      "vendor": "py-pdf",
      "product": "pypdf",
      "versions": [
        "2.10.5"
      ]
    }
  ],
  "Reference": [
    "https://github.com/py-pdf/pypdf/security/advisories/GHSA-hm9v-vj3r-r55m",
    "https://github.com/py-pdf/pypdf/issues/1329",
    "https://github.com/py-pdf/pypdf/pull/1331"
  ]
}